### PR TITLE
Disable kubectl download mirror options when not available

### LINKED ISSUE #3814: Azure mirror does not have kubectl binaries for MacOS
Since Azure mirror does not have kubectl binaries for MacOS, we probably should disable that option on MacOS

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes #3814

### CODE REVIEW

File: src/common/user-store/preferences-helpers.ts [hunk 1]
```diff
@@ -289,3 +309,8 @@ export const DESCRIPTORS = {
   editorConfiguration,
   terminalCopyOnSelect,
 };
+
+export const CONSTANTS = {
```
Comment (aleksfront) [inline src/common/user-store/preferences-helpers.ts]: Maybe it's just, but it feels like an old-schoolish way.

Why not export each of the constants separately and import them like this:

```
import { defaultPackageMirror, packageMirrors } from "../common/user-store/preferences-helpers";
```

Comment (Nokel81) [reply src/common/user-store/preferences-helpers.ts]: okay can do

Comment (Nokel81) [reply src/common/user-store/preferences-helpers.ts]: I want the descriptors to be separate because of https://github.com/lensapp/lens/pull/3292

**Review by aleksfront — DISMISSED** [review]
LGTM

**Pull request merged**